cmd/test-driver/main.go

```diff
@@ -53,33 +53,17 @@ func main() {
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
-	sourceManager := createSourceManagerOrDie(opt.Sources, opt.ScrapeTimeout)
+	sourceManager := sources.NewSourceManager(opt.Sources)
```

Comment:
Have `NewSourceManager` return an error if no sources are provided or if after parsing, the number of valid sources is 0? Maybe retain the `crateSourceManagerOrDie(...)` method and do error handling within that.